magma/primitives/when.py

```diff
@@ -76,17 +76,16 @@ def _get_corresponding_register_default(value: Type) -> Optional[Type]:
           current value)
         * For reg.CE, the default will be False (default to no update)
 
-    CE is assumed to be of type `Enable`
     """
     root_ref = value.name.root()
     try:
         value_inst = root_ref.inst
     except AttributeError:
         return None
-    if not isinstance(type(value_inst), Register):
+    if not type(value_inst)._magma_register_:
         return None
-    if isinstance(value, Enable):
-        return Enable(False)
+    if value is value_inst.get_enable():
+        return Bit(False)
```

Comment:
With the generic `get_enable API`, we have a more reliable way to find the enable values, but to make things more extensible (e.g. users that use a Bit enable type), we change the default wiring type to Bit.